src/com/senlainc/warsaw/tyurin/Turret.java

```diff
@@ -0,0 +1,9 @@
+package com.senlainc.warsaw.tyurin;
+
+public class Turret implements ILineStep, IProductPart{
```

Comment:
It's bad design. Object should not be implemented so different interfaces. It's like Turret at the same time assembly line step and part of the product, which this line should crete